packages/common-all/src/dnode.ts

```diff
@@ -1082,29 +1082,49 @@ export class NoteUtils {
   }
 
   /**
-   * Add {@link DNodeProps["parent"]}, {@link DNodeProps["children"]}, and backlinks from {@link noteHydrated} to {@link noteRaw}
+   * Add derived metadata from `noteHydrated` to `noteRaw`
+   * By default, include the following properties:
+   *  - parent
+   *  - children
+   *  - backLinks
    * @param noteRaw - note for other fields
-   * @param noteHydrated - note to get {@link DNodeProps["parent"]}, {@link DNodeProps["children"]}, and backlink properties from
+   * @param noteHydrated - note to get metadata properties from
    * @returns Merged Note object
    */
   static hydrate({
     noteRaw,
     noteHydrated,
+    opts,
   }: {
     noteRaw: NoteProps;
     noteHydrated: NoteProps;
+    opts?: Partial<{
+      keepAllLinks: boolean;
+      keepBackLinks: boolean;
+      keepAnchors: boolean;
+    }>;
   }) {
     const hydrateProps = _.pick(noteHydrated, ["parent", "children"]);
-    if (noteHydrated) {
-      noteRaw.links = noteHydrated.links.filter(
-        (link) => link.type === "backlink"
+
+    // check if we hydrate with links
+    if (opts?.keepAllLinks) {
```

Comment:
Couple questions. 
1. Are `keepAllLinks` and `keepAnchors` being used anywhere?
2. Any particular reason why we concat backlinks to noteRaw.links?

Reply:
not right now. originally, thought we could skip a call to update metadata by just keeping all links but inspected the code and realize that wasn't the case. kept the options here in case we wanted to use it down the line

as for why we concat - the idea is so we can chain links (eg. imagine hydrating with backlinks and frontmatter links) in which case we wouldn't want to wipe out what was in links

Reply:
Gotcha. The only thing I'd advise keeping on eye out is in case `noteRaw` already contains backlinks in which case we could have duplicates of the same thing. I remember running into it awhile back but I forgot which scenario caused it (or even if its possible anymore)

Reply:
got it. i removed the un-used options and reverted the logic to simplify things